[BUG] `ClearSky` fix missing value problem after transform

ciaran-g opened the pull request:
#### What does this implement/fix? Explain your changes.
<!--
A clear and concise description of what you have implemented.
-->

Missing values which should be retained as missing in the transformed domain were being overwritten to zero.

* Simple fix to only set transformed values to zero when they are below the clear sky power threshold **and** if the original value is not null

fkiraly reviewed (changes requested):
makes sense to me.

Would it be easy to add a test to certify for the fix?
(or modify an existing one, e.g., checking the output in a special case and expected values vs produced)

ciaran-g commented:
> Would it be easy to add a test to certify for the fix?

Added this franz 👍

Also added the author/function name in `sktime/datasets/_single_problem_loaders.py` which I just realised I forgot to add at the time

fkiraly reviewed (approved):
Great, thanks!
